openslides/core/static/js/core/projector.js

```diff
@@ -157,17 +157,57 @@ angular.module('OpenSlidesApp.core.projector', ['OpenSlidesApp.core'])
         $scope.broadcast = 0;
 
         var setElements = function (projector) {
-            $scope.elements = [];
+            // Get all elements, that should be projected.
+            var newElements = [];
             _.forEach(slides.getElements(projector), function (element) {
                 if (!element.error) {
                     // Exclude the clock if it should be disabled.
                     if (Config.get('projector_enable_clock').value || element.name !== 'core/clock') {
-                        $scope.elements.push(element);
+                        newElements.push(element);
                     }
                 } else {
                     console.error("Error for slide " + element.name + ": " + element.error);
                 }
             });
+
+            // Now we have to align $scope.elements to newElements:
+            // We cannot just assign them, because the ng-repeat would reload every
+            // element. This should be prevented (see #3259). To change $scope.elements:
+            // 1) remove all elements from scope, that are not in newElements (compared by the uuid)
+            // 2) Every new element in newElements, that is not in $scope.elements, get inserted there.
+            // 3) If there is the same element in newElements and $scope.elements every changed property
+            //    is copied from the new element to the scope element.
+
+            $scope.elements = _.filter($scope.elements, function (element) {
+                var found = false;
+                _.forEach(newElements, function (newElement) {
+                    if (element.uuid === newElement.uuid) {
+                        found = true;
+                    }
+                });
+                return found;
+            });
+
+            _.forEach(newElements, function (newElement) {
+                var found = false;
+                _.forEach($scope.elements, function (element) {
+                    if (element.uuid === newElement.uuid) {
+                        found = true;
+
+                        // copy all changed properties.
+                        _.forEach(newElement, function (value, key) {
```

Comment:
This will not work, because we do not want to copy angular stuff (e.g. the $hashCode changes every time).